urllib3/connection.py

```diff
@@ -470,7 +466,7 @@ class HTTPConnection(object):
     response_class = OldHTTPResponse
 
     def __init__(self, host, port, timeout=socket._GLOBAL_DEFAULT_TIMEOUT,
-                 source_address=None, socket_options=_UNKNOWN):
+                 source_address=None, socket_options=default_socket_options):
```

Comment:
Let's change this patch. For various boring reasons it's a bad idea to use mutable objects as the default values of function arguments: it doesn't do what you expect and can lead to really weird bugs. Better to use an `if is not None` check rather than throw it away entirely.